File: doc/examples/pdf/test/CalleeControllerTest.java
package org.goafabric.calleeservice.controller;

import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Test;

import java.util.Base64;

import static io.restassured.RestAssured.given;

@QuarkusTest
class CalleeControllerTest {

    @Test
    public void sayMyName() {
        given()
                .auth().basic(new String(Base64.getDecoder().decode("YWRtaW4=")), new String(Base64.getDecoder().decode("YWRtaW4=")))
                .when().get("/callees/sayMyName?name=Heisenberg")
                .then()
                .statusCode(200);
    }

    @Test
    public void sayMyOtherName() {
        given()
                .auth().basic(new String(Base64.getDecoder().decode("YWRtaW4=")), new String(Base64.getDecoder().decode("YWRtaW4=")))
                .when().get("/callees/sayMyOtherName/Andreas")
                .then()
                    .statusCode(200);
    }

    @Test
    public void createPdf() {
        given()
                .auth().basic(new String(Base64.getDecoder().decode("YWRtaW4=")), new String(Base64.getDecoder().decode("YWRtaW4=")))
                .when().get("/pdfs/createpdf")
                .then()
                .statusCode(200);
    }

}

File: src/test/kotlin/org/goafabric/calleeservice/controller/CalleeControllerTest.kt
package org.goafabric.calleeservice.controller

import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured
import org.junit.jupiter.api.Test
import java.util.*

@QuarkusTest
open class CalleeControllerTest {
    @Test
    fun sayMyName() {
        RestAssured.given()
            .`when`().get("/callees/sayMyName?name=Heisenberg")
            .then()
            .statusCode(200)
    }

    @Test
    fun sayMyOtherName() {
        RestAssured.given()
            .`when`().get("/callees/sayMyOtherName/Andreas")
            .then()
            .statusCode(200)
    }
}